core/lib/src/config/config.rs

```diff
@@ -49,6 +50,7 @@ pub struct Config {
     /// How much information to log.
     pub log_level: LoggingLevel,
     /// The secret key.
+    #[cfg(feature = "private-cookies")]
```

Comment:
I think the possibility was discussed on IRC of leaving `secret_key` in regardless of `cfg`. I realize that `cookie` only exposes SecretKey when `secure` is enabled, but I think `SecretKey` could become a unit struct in the alternative case if reducing the `cfg` annotations is an important goal. I suppose that might have downsides, though.